documentation/src/docs/state-synchronizer-design.md

```diff
@@ -7,24 +7,28 @@ You may obtain a copy of the License at
 
 # Example
-A concrete example synchroning the contents of a set is [checked in here](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/SetSynchronizer.java). We also have an example that is synchronizing [membership of a set of hosts](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/MembershipSynchronizer.java).
 
-Imagine you want many processes to share a Map. This can be done by creating by a StateSynchronizer, which will help coordinate changes to the map. Each client has their own copy of the map in memory and can apply updates by passing a generator to the StateSynchronizer. Every time an update is attempted the updated is recorded to a segment. Updates will fail unless the Map passed into the update method is consistent with all of the updates that have been recorded to the segment. If this occurs the generator is called with the latest state to try again. Thus the order of updates is defined by the order in which they are written to the segment.
+A concrete [example](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/SetSynchronizer.java) of synchronizing the contents of a set is provided. We also have an example that is synchronizing [membership of a set of hosts](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/MembershipSynchronizer.java).
 
-# How this is implemented
-For this to work we use two features of the Pravega Segment Store Service.
-## Conditional append
-The append method can specify what offset the append is expected to be at. If the append will not go in, nothing should be done and a failure should be returned to the client.
+Imagine you want many processes to share a Map. This can be done by creating the State Synchronizer, it will aid in coordinating the changes to the Map. Each client has their own copy of the Map in memory and can apply updates by passing a generator to the State Synchronizer. Every time an update is attempted the updates are recorded to the Stream Segment. Updates are successful when the Map passed into the update method is consistent with all of the updates that have been recorded to the Stream Segment. If this occurs the generator is called with the latest state to try again. Thus the order of updates is defined by the order in which they are written to the Stream Segment.
 
-## Truncate segment
-Truncate segment deletes all data before a given offset. (This operation does not affect the existing offsets)
-Any reads for offsets lower than this value will fail. Any data stored below this offset can be removed.
+# Implementation
+
+For the implementation, two features of the Pravega Segment Store Service is used.
+
+## Conditional Append
```

Comment:
The text of both sub-sections is not linked with the intro (i.e., first sentence). In fact, these subsection could be a bit more self-explanatory, like: `The conditional append call in the Pravega Segment Store is the cornerstone for the implementation of the State Synchronizer semantics. That is, when a client updates a piece of data via State Synchronizer, a conditional append is internally used against the Segment Store. In a conditional append, the client specifies the offset in which the append is expected to be at. If the offset provided by the client does match the actual offset of the append in the Stream Segment, the operation is aborted and an error is returned to the client. This mechanism is exploited in the State Synchronizer to provide optimistic locks on data updates.`

Reply:
That should be "in which the append is expected to be located" not "at". And we should say "used" not "exploited".

Reply:
@tkaitchuck Accepted suggestion